org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/scan/internal/DefaultBuildScanRegistry.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2017 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.scan.internal;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.base.Optional;
+
+import org.eclipse.buildship.core.console.ProcessDescription;
+import org.eclipse.buildship.core.scan.BuildScanRegistry;
+
+/**
+ * Default implementation for {@link BuildScanRegistry}.
+ *
+ * @author Donat Csikos
+ */
+public final class DefaultBuildScanRegistry implements BuildScanRegistry {
+
+    private final Map<ProcessDescription, String> buildScans = Collections.synchronizedMap(new HashMap<ProcessDescription, String>());
```

Comment:
We can't have a service that grows indefinitely. The information about the last build scan should instead be stored somewhere in the execution page.